File: recursion/python/concat_array.py
# concatenate elements in an array of strings

def concat_array_iterative(arr):
    string = ''

    for i in arr:
        string += i

    return string


def concat_array_recursive(arr):
    if len(arr) == 0:
        return ''
    else:
        head = arr[0]
        tail = arr[1:]
        return head + concat_array_recursive(tail)


if __name__ == '__main__':
    print(concat_array_iterative(['this', 'is', 'a', 'test']))
    print(concat_array_recursive(['this', 'is', 'a', 'test']))

File: recursion/javascript/concatArray.js
// concatenate elements in an array of strings

function concatArrayIterative(arr) {
    let string = '';

    for (val of arr) {
        string += val;
    }

    return string;
}

function concatArrayRecursive(arr) {
    if (arr.length === 0) {
        return '';
    } else {
        let head = arr[0];
        let tail = arr.slice(1, arr.length);

        return head + concatArrayRecursive(tail);
    }
}

console.log(concatArrayIterative(['this', 'is', 'a', 'test']));
console.log(concatArrayRecursive(['this', 'is', 'a', 'test']));
